### PR TITLE
Document mimirtool backfill command.

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR adds documentation for `mimirtool backfill` command.

#### Checklist

- [na] Tests updated
- [x] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 1]
```diff
@@ -872,6 +876,38 @@ printf "{\n%s\n}" "${key_values::-1}"
 
 The only parameter of the script is a file containing the flags, with each flag on its own line.
 
+### Backfill
+
+Backfill command uploads Prometheus TSDB blocks into Grafana Mimir, by using block upload API exposed by compactor component.
+
+If the command is interrupted it can be started again. Mimirtool will detect which blocks are already uploaded, and will only upload unfinished or new blocks.
+
+If block upload is not enabled for the user, `mimirtool backfill` will fail.
```
Comment (pracucci) [inline docs/sources/operators-guide/tools/mimirtool.md:885]: I would rephrase it, saying as first thing that block upload is disabled by default. Then how to enable it (link to config doc page), and finally what happens if you try to use it with `mimirtool` but it's disabled (even if I guess at that point it will be obvious).

**Review by pracucci — APPROVED** [review]
Good job! I just left a couple of nits. Please remember to ping docs squad for a review.

**Comment by pstibrany** [comment]
> Please remember to ping docs squad for a review.

I thought adding `type/docs` label does the trick. Should I always ask @osg-grafana for review?

File: docs/sources/operators-guide/tools/mimirtool.md (hunk 1 above)
Comment (pstibrany) [reply docs/sources/operators-guide/tools/mimirtool.md:887]: Thanks, done.

**Comment by pracucci** [comment]
Closes #2383

File: docs/sources/operators-guide/configure/configure-tsdb-block-upload.md [hunk 2]
```diff
@@ -13,6 +13,8 @@ Grafana Mimir itself.
 For information about limitations that relate to importing blocks from Thanos, see
 [Migrating from Thanos or Prometheus to Grafana Mimir]({{< relref "../../migration-guide/migrating-from-thanos-or-prometheus.md" >}}).
 
+To upload blocks, refer to [mimirtool backfill]({{< relref "../tools/mimirtool.md#backfill" >}}) command.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/configure/configure-tsdb-block-upload.md:16]:
```suggestion
To upload blocks, refer to the [mimirtool backfill]({{< relref "../tools/mimirtool.md#backfill" >}}) command.
```

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 3]
```diff
@@ -872,6 +876,39 @@ printf "{\n%s\n}" "${key_values::-1}"
 
 The only parameter of the script is a file containing the flags, with each flag on its own line.
 
+### Backfill
+
+Backfill command uploads Prometheus TSDB blocks into Grafana Mimir, by using block upload API exposed by compactor component.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:881]:
```suggestion
The `backfill` command uploads Prometheus TSDB blocks into Grafana Mimir, by using the block-upload API that is exposed by the compactor component.
```
There are several function words missing from this sentence that I filled in. Please vet for technical accuracy.

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 4]
```diff
@@ -872,6 +876,39 @@ printf "{\n%s\n}" "${key_values::-1}"
...
+To enable block upload feature for user or entire system, refer to [Configure TSDB block upload]({{< relref "../configure/configure-tsdb-block-upload.md" >}}).
+If block upload is not enabled for the user, `mimirtool backfill` will fail.
+
+##### Example
+
+```bash
+mimirtool backfill --address=http://mimir-compactor/ --id=anonymous /var/prometheus/{01G803NFXZ0MVKN71GT91HMV3Z,01G8BQ8PRR4TAP7EXZVBNTRBZ4,01G8CB7GTTC5ZXY23WTXHSYQXQ}
+```
+
+Running the command results in the following output:
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:895]:
```suggestion
The results of the `backfill` command are as follows:
```

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 5]
```diff
@@ -872,6 +876,39 @@ printf "{\n%s\n}" "${key_values::-1}"
...
+INFO[0000] making request to start block upload     block=01G8CB7GTTC5ZXY23WTXHSYQXQ file=meta.json path=/var/prometheus/01G8CB7GTTC5ZXY23WTXHSYQXQ
+INFO[0000] uploading block file                     block=01G8CB7GTTC5ZXY23WTXHSYQXQ file=index path=/var/prometheus/01G8CB7GTTC5ZXY23WTXHSYQXQ size=151181
+INFO[0000] uploading block file                     block=01G8CB7GTTC5ZXY23WTXHSYQXQ file=chunks/000001 path=/var/prometheus/01G8CB7GTTC5ZXY23WTXHSYQXQ size=1986792
+INFO[0000] block uploaded successfully              block=01G8CB7GTTC5ZXY23WTXHSYQXQ path=/var/prometheus/01G8CB7GTTC5ZXY23WTXHSYQXQ
+INFO[0001] finished uploading blocks                already_exists=1 failed=0 succeeded=2
+```
+
 ## License
 
 Licensed AGPLv3, see [LICENSE](https://github.com/grafana/mimir/blob/main/LICENSE).
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:914]:
```suggestion
This software is licensed as AGPLv3. For more information, see [LICENSE](https://github.com/grafana/mimir/blob/main/LICENSE).
```
@brendamuir, do we have boilerplate text for license references?

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 6]
```diff
@@ -872,6 +876,39 @@ printf "{\n%s\n}" "${key_values::-1}"
 
 The only parameter of the script is a file containing the flags, with each flag on its own line.
 
+### Backfill
+
+Backfill command uploads Prometheus TSDB blocks into Grafana Mimir, by using block upload API exposed by compactor component.
+
+If the command is interrupted it can be started again. Mimirtool will detect which blocks are already uploaded, and will only upload unfinished or new blocks.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:883]:
```suggestion
If the command is interrupted, you can restart it. Mimirtool detects which blocks are already uploaded, and will only upload unfinished or new blocks.
```

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 7]
```diff
@@ -872,6 +876,39 @@ printf "{\n%s\n}" "${key_values::-1}"
...
+
+Block upload feature is disabled by default.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:885]:
```suggestion
The block-upload feature is disabled by default.
```

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 8]
```diff
@@ -872,6 +876,39 @@ printf "{\n%s\n}" "${key_values::-1}"
...
+To enable block upload feature for user or entire system, refer to [Configure TSDB block upload]({{< relref "../configure/configure-tsdb-block-upload.md" >}}).
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:886]:
```suggestion
To enable the block-upload feature for a user or an entire system, refer to [Configure TSDB block upload]({{< relref "../configure/configure-tsdb-block-upload.md" >}}).
```

**Review by osg-grafana — APPROVED** [review]
Unblocking with feedback

**Comment by osg-grafana** [comment]
> > Please remember to ping docs squad for a review.
> 
> I thought adding `type/docs` label does the trick. Should I always ask @osg-grafana for review?

If you add me as a reviewer, I can search for "Awaiting review from me" in the queue and get to things more quickly.

**Comment by pstibrany** [comment]
> Unblocking with feedback

@osg-grafana Thanks for providing suggestions and making my life easy by simply committing them.
